### PR TITLE
Tree generator

### PR DESCRIPTION
**Pull request by pancudaniel7**
- Tree generator added
-  Observable pattern

### CODE REVIEW

File: patterns/Observer.kt [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+class GardenListener : Listener {
+    override fun update(action: String) {
+        println("Garden listener $action\n")
+    }
+}
+
+class HouseListener : Listener {
+    override fun update(action: String) {
+        println("House listener $action")
```
Comment (lradu-godaddy) [inline patterns/Observer.kt:56]: Trivial: noticed that for Garden listener you are printing `\n` at the end of line, maybe you need this for house and car listeners as well.

**Review by lradu-godaddy — APPROVED** [review]

File: patterns/Observer.kt (hunk 1 above)
Comment (pancudaniel7) [reply patterns/Observer.kt:56]: Good point, I will make the changes 👍

File: tree/java/TreeGenerator.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+        waitForExecutorToFinish(5);
+    }
+
+    public static void generateRandomTree(Node root) {
+        generateRandomBasicTree(root, 2, 3, atomicHeight.get());
+    }
+
+    public static void generateRandomBinaryTree(Node root) {
+        generateRandomBasicTree(root, 2, 10, atomicHeight.get());
```
Comment (danieljecu) [inline tree/java/TreeGenerator.java:28]: why 10?

Comment (danieljecu) [reply tree/java/TreeGenerator.java:28]: oh is actually a fixed numer of nodes

**Review by danieljecu — APPROVED** [review]
LGTM, i see you also use kotlin 👍

**Pull request merged**